pkg/volume/emptydir/empty_dir.go

```diff
@@ -107,17 +110,58 @@ func (plugin *emptyDirPlugin) NewMounter(spec *volume.Spec, pod *v1.Pod, opts vo
 	return plugin.newMounterInternal(spec, pod, plugin.host.GetMounter(plugin.GetPluginName()), &realMountDetector{plugin.host.GetMounter(plugin.GetPluginName())}, opts)
 }
 
+func calculateEmptyDirMemorySize(nodeAllocatableMemory *resource.Quantity, spec *volume.Spec, pod *v1.Pod) *resource.Quantity {
+	// if feature is disabled, continue the default behavior of linux host default
+	sizeLimit := &resource.Quantity{}
+	if !utilfeature.DefaultFeatureGate.Enabled(features.SizeMemoryBackedVolumes) {
+		return sizeLimit
+	}
+
+	// size limit defaults to node allocatable (pods cant consume more memory than all pods)
+	sizeLimit = nodeAllocatableMemory
+	zero := resource.MustParse("0")
+
+	// determine pod resource allocation
+	// we use the same function for pod cgroup assigment to maintain consistent behavior
+	// NOTE: this could be nil on systems that do not support pod memory containment (i.e. windows)
+	podResourceConfig := cm.ResourceConfigForPod(pod, false, uint64(100000))
+	if podResourceConfig != nil && podResourceConfig.Memory != nil {
+		podMemoryLimit := resource.NewQuantity(*(podResourceConfig.Memory), resource.BinarySI)
+		// ensure 0 < value < size
+		if podMemoryLimit.Cmp(zero) > 0 && podMemoryLimit.Cmp(*sizeLimit) < 1 {
+			sizeLimit = podMemoryLimit
+		}
+	}
+
+	// volume local size is  used if and only if less than what pod could consume
+	if spec.Volume.EmptyDir.SizeLimit != nil {
+		volumeSizeLimit := spec.Volume.EmptyDir.SizeLimit
+		// ensure 0 < value < size
+		if volumeSizeLimit.Cmp(zero) > 0 && volumeSizeLimit.Cmp(*sizeLimit) < 1 {
```

Comment:
I know this is just alpha so it's fine if you plan on tackling this later, but I think that there should be some sort of indicator to the user if they try to use a size limit in their config that's bigger than what the size limit actually is. At the very least something should be logged, though I think either the pod should be not allowed to be admitted or an event should be created.

Reply:
@2rs2ts maybe we can throw an event.  the pod author is unlikely to have access to node logs in many contexts.  we can explore that after this lands.

Reply:
>we can explore that after this lands

Fine by me.

>the pod author is unlikely to have access to node logs in many contexts

Yes, that'll be the case for most of my coworkers 😅 as a cluster admin I would want logs at a minimum because I could write a scheduled search in our log storage to detect if it happened to anybody, and reach out to them. From a user perspective, events are visible, but they don't get shoved in their face, so I think most of my colleagues would miss the event. So blocking admission would feel like the best way to make sure my users know this is happening, but as long as there's some sort of indicator, rather than this silent change, I'd be able to adopt the feature